### PR TITLE
Let exits use consolidation queue (EIP-8080)

### PR DESCRIPTION
**Pull request by barnabasbusa**
## Summary

Implements [EIP-8080](https://eips.ethereum.org/EIPS/eip-8080) ("Let exits use the consolidation queue") at the Gloas fork.

### Spec change

Single function override in `specs/gloas/beacon-chain.md` — **Modified `compute_exit_epoch_and_update_churn`**. Adds a routing prefix that checks whether the exit queue has grown longer than the consolidation queue; if so, the exit is satisfied through `compute_consolidation_epoch_and_update_churn` with `exit_balance` converted by the `2/3` factor that reflects the relative weak-subjectivity weight of exit vs. consolidation churn.

Intro bullet list updated to reference EIP-8080.

No new constants, no new state fields, no fork transition logic.

### Composition with EIP-8061 (#5061)

This PR is written against current `master`, so the inner churn-limit call in the preserved tail of `compute_exit_epoch_and_update_churn` is still `get_activation_exit_churn_limit(state)` (the Electra form). If #5061 lands first, this PR needs a one-line update in the same function to switch to `get_exit_churn_limit(state)` — mechanical rebase, no semantic change.

The EIP-8080 addition and the EIP-8061 rewrite are orthogonal: 8080 adds a routing prefix, 8061 changes the churn-limit accessor in the tail. They compose cleanly regardless of merge order.

### Design notes

- The comparison uses the *adjusted* `earliest_exit_epoch` / `earliest_consolidation_epoch` (both `max()`-clamped to `activation_exit_epoch`), matching the ["full function" form](https://eips.ethereum.org/EIPS/eip-8080) in the EIP and the correction in [ethereum/EIPs#10883](https://github.com/ethereum/EIPs/pull/10883).
- The `Gwei(2 * exit_balance // 3)` wrap is required by the mypy type check (the `Gwei` typed constant is not auto-coerced from the divided `int`).
- Weak-subjectivity budget is unchanged — one unit of exit churn corresponds to `2/3` of a unit of consolidation churn, so the total weak-subjectivity impact is preserved whether consolidation churn is spent on consolidations or on rerouted exits.

## Tests

Six unit tests under `tests/core/pyspec/eth_consensus_specs/test/gloas/unittests/test_compute_exit_epoch_and_update_churn.py`:

- [x] No routing when exit and consolidation queues are equal length
- [x] No routing when consolidation queue is longer than exit queue
- [x] Routes through consolidation queue when exit queue is longer
- [x] `2/3` conversion factor applied to `exit_balance` when routed
- [x] One-epoch difference is enough to trigger routing (edge case)
- [x] Exit-queue state fields (`earliest_exit_epoch`, `exit_balance_to_consume`) are preserved across a routing call

Local verification: `make lint` clean, all 6 tests pass on the `minimal` preset.

## Test plan

- [ ] CI passes on all presets
- [ ] Rebase onto #5061 (or vice versa) once either PR lands and sync the `per_epoch_churn` accessor accordingly
- [ ] Cross-client review, especially around the interaction with `earliest_consolidation_epoch` bookkeeping when exits and consolidations arrive in the same block

## Related

- EIP text: https://eips.ethereum.org/EIPS/eip-8080
- EIP spec-inconsistency fix: ethereum/EIPs#10883
- EIP test-case additions: ethereum/EIPs#11290
- Composes with #5061 (EIP-8061: increase exit and consolidation churn)

### CODE REVIEW

File: specs/gloas/beacon-chain.md [hunk 1]
```diff
@@ -785,6 +788,48 @@ def initiate_builder_exit(state: BeaconState, builder_index: BuilderIndex) -> No
...
+queue, converting the requested `exit_balance` by the `2/3` factor that reflects
+the relative weak-subjectivity weight of exit vs. consolidation churn.
+
+```python
+def compute_exit_epoch_and_update_churn(state: BeaconState, exit_balance: Gwei) -> Epoch:
+    activation_exit_epoch = compute_activation_exit_epoch(get_current_epoch(state))
+    earliest_exit_epoch = max(state.earliest_exit_epoch, activation_exit_epoch)
+
+    # Route exits through the consolidation queue when it is shorter
+    # [New in Gloas:EIP8080]
```
Comment (jtraglia) [inline specs/gloas/beacon-chain.md]: The "new in" comment should go above.
```suggestion
    # [New in Gloas:EIP8080]
    # Route exits through the consolidation queue when it is shorter
```

File: tests/core/pyspec/eth_consensus_specs/test/gloas/block_processing/test_compute_exit_epoch_and_update_churn.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    pre_consolidation_epoch = state.earliest_consolidation_epoch
+
+    exit_balance = spec.Gwei(1 * 10**9)
+    returned_epoch = spec.compute_exit_epoch_and_update_churn(state, exit_balance)
+
+    assert returned_epoch == state.earliest_exit_epoch
+    assert state.earliest_consolidation_epoch == pre_consolidation_epoch
+    assert state.consolidation_balance_to_consume == pre_consolidation_balance
+    assert state.earliest_exit_epoch >= spec.Epoch(activation_exit_epoch + 3)
+    yield "post", state
```
Comment (jtraglia) [inline tests/core/pyspec/eth_consensus_specs/test/gloas/block_processing/test_compute_exit_epoch_and_update_churn.py:41]: I noticed that we're not yielding a "pre" state, but since this is a unit test we're not supposed to yield anything. Regardless, we should convert all of these to reference tests (not unit tests) so that clients can use them.

**Comment by barnabasbusa** [comment]
No longer needed
